### PR TITLE
Fix building on OSX if Unity’s within /Applications/Unity/Unity.app

### PR DESCRIPTION
**Pull request by ghost**
If `/Applications/Unity/Unity.app` exists, we can use that one.

Q: do we / should we validate it matches projectVersion?

### CODE REVIEW

**Comment by jessicafalk** [comment]
Can we validate?

**Comment by ghost** [comment]
Depends... it's not done for Windows, so I can add "validate project version match" as a feature in this/another PR

**Comment by DanailPenev** [comment]
If it's not done for Windows either, I would say validating is a job for a different PR.

**Comment by jessicafalk** [comment]
agreed

**Comment by jamiebrynes7** [comment]
Validating the Unity version is surprisingly hard. There's no `--version` switch.

File: tools/UnityPaths/UnityPaths.cs [hunk 1]
```diff
@@ -53,6 +53,11 @@ public static string TryGetUnityPath()
                 return hubPath;
             }
 
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX) && Directory.Exists(Path.Combine(DefaultUnityPath, "Unity.App")))
```
Comment (gkassabli) [inline tools/UnityPaths/UnityPaths.cs:56]: I just installed Unity with the Hub on my Mac and the path it is installed to is /Applications/Unity/Hub/Editor/_version_/Unity.app
Do we want to cover that scenario?

**Comment by jessicafalk** [comment]
yeah, that's what I thought

File: tools/UnityPaths/UnityPaths.cs (hunk 1 above)
Comment (jessicafalk) [reply tools/UnityPaths/UnityPaths.cs:56]: that is already covered afaik

Comment (gkassabli) [reply tools/UnityPaths/UnityPaths.cs:56]: Okay, reading the code, I guess that should be covered by the code above. If that is the case, ignore my comment

Comment (ghost) [reply tools/UnityPaths/UnityPaths.cs:56]: Aye, if it's the right version it should

**Review by jessicafalk — APPROVED** [review]

**Review by DanailPenev — APPROVED** [review]

**Pull request merged**